### ISSUE TITLE
Design problem: the envvar HOST is typically set on Linux systems

### ISSUE THREAD

**Issue by tecosaur** [GenieFramework/Genie.jl#453]
**Describe the bug**

After loading the environment settings, Genie checks the `ENV["HOST"]` key and overrides the environment set `server_host` if a value is given.

https://github.com/GenieFramework/Genie.jl/blob/543b630927e01de48511aa5e1eefc1243f301180/src/App.jl#L28

This is problematic on Linux systems where the `HOST` environment variable is typically set to the device's hostname. Fox example, on my system:
```sh
~$ sh -c 'echo $HOST'
tranquillity
```

This is of course inherited by Julia, and so `ENV["HOST"]` is always set and the environment set `server_host` always ignored.

**Further comments**

This code reads as if it is intended for use in overrides like
```
HOST=1.2.3.4 ./bin/server
```

This has a similar problem to using `USER`, or any other commonly system-set variable (`SHELL`, `LANGUAGE`, `HOME`, etc.).
IMO the clear fix for this is to ensure the variables are Genie-specific by adding an appropriate prefix.
For instance, I'd change the current use of `PORT`, `WSPORT`, and `HOST` to `GENIE_PORT`, `GENIE_WSPORT`, and `GENIE_HOST`. This would be a breaking change, but any fix for this name conflict would have to be.

This is also a rather standard way of doing things, see `JAVA_HOME`, `KDE_HOME`, and `DOCKER_CONFIG` for instance.

**Aditional context**

Given this is a issue with the code logic, I think my system isn't relevant here.

**Comment by essenciary** [comment]
Thanks for reporting this. A non breaking solution might also be to make Genie handle the default HOST (eg "tranquility"). I'll take a look - it's not a common issue as it's really the first time I see it, so it's quite strange.

**Comment by tecosaur** [comment]
I'm surprised this hasn't come up before. It really isn't that rare. Some linux distros set it, and in fact you can find `HOST` mentioned in old Linux/Unix material (see https://projects.ncsu.edu/hpc/Documents/unixtut/unix8.html, from 1995).

A number of shells also set `HOST` during initialisation, like **zsh** (as of zsh 2, released in 1991.
You can find this in the current code base fairly easily, see: https://github.com/zsh-users/zsh/blob/c6a85163619ed1cee89ab047a0d98108ed46828d/Src/params.c#L841).
**tcsh** has been using `HOST` since 1987 (https://groups.google.com/g/comp.sources.unix/c/v-q1qRVIpi4/m/TbAvM_gFD-ge), **csh** also seems to set `HOST`, etc.

To me zsh alone is reason enough to change this — it's in all likelihood the second most popular shell on Linux, and has replaced bash on MacOS (as of 2019). Even if no one has reported this before now, it *will* trip up a lot of users.

Yes, no longer looking at `HOST` is a breaking change, but I don't think there's an easy way around it. Beyond the specific issues with `HOST` here, relying on unprefixed terms with common usage is asking for trouble. That's why every major application I've looked at prefixes the env vars they use (e.g. `JAVA_`, `KDE_`, `DOCKER_`, etc.), and I think it behoves Genie to do the same.

**Comment by essenciary** [comment]
Yes, I agree... It's just interesting that it hasn't really occurred until now (I've been running code on Linux, WSL2, macOS without any problems). What Linux distro do you have? For instance, zsh does not set HOST on macos:

<img width="657" alt="image" src="https://user-images.githubusercontent.com/11292378/146801174-4192ad05-388b-4613-a15f-45e1afeefc54.png">

**Comment by essenciary** [comment]
But yes, regardless, prefixing the env vars is definitely the best approach.

**Comment by tecosaur** [comment]
I'm not exactly sure what's going on with that example you have, from your `echo $HOST` line it looks like it's set to me, but not passed on to the Julia process? ( IMO this inconsistency is a good reason in and of itself for changing behaviour, any `HOST`-affected troubleshooting is liable to be a pain )

For what it's worth, I'm using openSUSE Tumpleweed, so my experience likely directly translates to openSUSE Leap and SUSE Linux Enterprise (which is probably the # 2 commercial Linux distro, behind RHEL).
